legal-api/src/legal_api/resources/nr_type_map.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Name Request Type mapping service.
+
+Provides an endpoint to query/match NR Type to Entity Type mappings.
+"""
+from enum import Enum
+from http import HTTPStatus
+
+from flask import request
+from flask_restx import Namespace, Resource
+
+from legal_api.models import Business
+
+
+API = Namespace('nrTypeMap', description='NR Type mapping service')
+
+
+class RequestActions(Enum):
+    """Render an Enum of the Request Actions. Should match RequestAction enum in namex project."""
+
+    ACHG = 'ACHG'
+    AML = 'AML'  # amalgamate
+    AS = 'AS'  # assumed
+    CHG = 'CHG'  # change (ie, alter)
+    CNV = 'CNV'  # convert
+    MVE = 'MVE'  # move
+    NEW = 'NEW'  # new business
+    NEW_AML = 'NEW_AML'  # new or amalgamate
+    REH = 'REH'
+    REN = 'REN'  # renew
+    REST = 'REST'  # restore
+
+
+@API.route('', methods=['GET'])
+@API.doc(params={
+    'nrType': {'description': 'Optional NR Type code.', 'example': 'CR'},
+    'entityType': {'description': 'Optional Entity Type code.', 'example': 'BC'},
+    'RequestActions': {'description': 'Optional Request Action code.', 'example': 'NEW'}
+})
+class NrTypeMap(Resource):
+    """NR Type mapping service."""
+
+    @staticmethod
+    def get():
+        """Return a list of mappings matching any query params."""
+        # get query params
+        nr_type_cd = (request.args.get('nrTypeCd', None).upper()) \
+            if request.args.get('nrTypeCd', None) else None
+        entity_type_cd = (request.args.get('entityTypeCd', None).upper()) \
+            if request.args.get('entityTypeCd', None) else None
+        request_action_cd = (request.args.get('requestActionCd', None).upper()) \
+            if request.args.get('requestActionCd', None) else None
+
+        # build list of tuples
+        # should match Request Type Mapping in namex project
+        # NB: commented out tuples that exist in namex but are not yet supported by Lear
+        request_type_mapping = [
+            # NR Type Code, Legal Type the NR type code is valid for, Filing Type we can do
+            ('CR', Business.LegalTypes.COMP.value, RequestActions.NEW_AML.value),
+            ('CR', Business.LegalTypes.COMP.value, RequestActions.NEW.value),
+            ('CR', Business.LegalTypes.COMP.value, RequestActions.AML.value),
+            ('CCR', Business.LegalTypes.COMP.value, RequestActions.CHG.value),
+            ('CT', Business.LegalTypes.COMP.value, RequestActions.MVE.value),
+            ('RCR', Business.LegalTypes.COMP.value, RequestActions.REST.value),
+            ('RCR', Business.LegalTypes.COMP.value, RequestActions.REH.value),
+            ('RCR', Business.LegalTypes.COMP.value, RequestActions.REN.value),
+            # ('XCR', Business.LegalTypes.XPRO_CORPORATION.value, RequestActions.NEW_AML.value),
```

Comment:
why are these commented out?

Reply:
Because they are valid mappings but Lear doesn't yet support the legal types they refer to. IMHO it's better to leave the mapping intact (ie, similar to its original project). These can simply be uncommented (and order preserved) when Lear supports the new legal types.

Reply:
I think this list will be used as a reference for what could be done but still every consumer will filter themselves to the legalTypes they want to deal with. 
E.g. [#6852](https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6852) will filter by the 3 legalTypes (CR, UL, CC) and check the options to affiliate these 3 types only.
Maybe @thorwolpert has something different in mind... thoughts? :)

Reply:
If all valid mappings should be returned then I will need to re-enable the Legal Types enums (even though Lear doesn't support them yet).

Let me know what you prefer. (I prefer returning all mappings.)

Reply:
@thorwolpert What do you think about adding a tuple value indicating whether Lear supports the current type?

This would allow the table to return all mappings, and would allow Sumesh (Relationships) to know which types are / not supported.

For example, `GET /nrTypeMap?nrTypeCd=XCR` would return the 3 matching items and a flag on each of them indicating "xpro corporation is not yet supported".